main.go

```diff
@@ -143,11 +157,77 @@ func watchGGST(noClose bool, ctx context.Context) {
 	}
 }
 
+func autoUpdate() {
+
+	url := GithubDownloadURL
+	if UngaBungaMode == "" {
+		url = GithubDownloadURL + UpdateName
+	} else {
+		url = GithubDownloadURL + UpdateNameUngaBunga
+	}
+
+	client := http.Client{
+		Timeout: UpdateTimeout,
+	}
+	resp, err := client.Get(GithubReleasesURL)
+
+	if err != nil {
+		fmt.Println("Couldn't get latest release version number")
+		return
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		fmt.Println("Couldn't read latest release version")
+		return
+	}
+
+	release := &Release{}
+	json.Unmarshal(body, release)
+
+	if release.TagName != Version && Version != "" {
```

Comment:
Sorry I actually missed this. The default string for `Version` is actually `(unknown version)` not empty string :P .

That being said, this still has the problem where it will downgrade pre-release versions.

Since I follow semver for the releases, you can use a semver library (eg. blang/semver) to parse the version number and check if the version on Github is actually newer.